### ISSUE TITLE
Error handling for Products release 1

### ISSUE THREAD

**Issue by jaclync** [woocommerce/woocommerce-ios#1652]
- [x] support inline error in `TitleAndTextFieldTableViewCell` https://github.com/woocommerce/woocommerce-ios/pull/1695
- [x] invalid SKU https://github.com/woocommerce/woocommerce-ios/pull/1648#issuecomment-568772727 --> moved to #1710 
- [x] Price settings: validate sale price is lower than the normal price #1749 
- [ ] Shipping settings: error when retrieving shipping class
- [ ] Price Settings: error when retrieving tax class

**Comment by pmusolino** [comment]
@jaclync 
For these two cases
> Shipping settings: error when retrieving shipping class
> Price Settings: error when retrieving tax class

also here we will show a Notice alert or a sort of empty status? I wasn't able to find any reference to this 👍 do you have any idea about it?

**Comment by jaclync** [comment]
@pmusolino for these two cases where the current shipping class / tax class isn't available, we might want to prevent them from updating the field by disabling the tap action. when we receive the sync error, we can also show an error Notice.

Right now we don't have a UX (like pull to refresh) to resync the shipping/tax class, maybe we can also add an action to the Notice alert to try syncing again? Just a thought 😄

**Comment by pmusolino** [comment]
> for these two cases where the current shipping class / tax class isn't available, we might want to prevent them from updating the field by disabling the tap action. when we receive the sync error, we can also show an error Notice.

What are the cases where they are not available? We always have the information. The only case where there is an external call is in pricing, where we try to get the name of the slug used in the "Tax Class". But if it fails, we show the slug (which is still representative of the information).

> Right now we don't have a UX (like pull to refresh) to resync the shipping/tax class, maybe we can also add an action to the Notice alert to try syncing again? Just a thought.

I agree with your idea, and looking at the implementation it's already managed like this, so if the sync fails, we show a Notice alert with the retry option. 

Basically, it seems that these two points have already been managed:
- [x] Shipping settings: error when retrieving shipping class
- [x] Price Settings: error when retrieving tax class

**Comment by pmusolino** [comment]
The other open points of this issue are already managed. Closing it now.
